src/btree/bt_split.c

```diff
@@ -2299,7 +2299,7 @@ __wt_split_reverse(WT_SESSION_IMPL *session, WT_REF *ref)
  *     Rewrite an in-memory page with a new version.
  */
 int
-__wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi)
+__wt_split_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, WT_MULTI *multi, bool change_ref_state)
```

Comment:
This new parameter could use a comment. How should the caller know to change the ref state?

Reply:
The caller knows if itself changes the state later.

Reply:
Right, but my point is that the ref state was previously abstracted away. This is no longer the case, so anyone calling `__wt_split_rewrite` has to think about the ref state, where they didn't before. Is it reasonable to design this some other way?

Reply:
See https://github.com/wiredtiger/wiredtiger/pull/11864#discussion_r2078898751